### DISCUSSION TITLE
Is possible workflow execute activity with different domain?

### DISCUSSION THREAD

**Discussion by zedongh** [uber/cadence#4907] [Q&A]
Using workflow.ExecuteActivity with different domain (with workflow) could be help for user test it's new version activity implementation and not affect it's production implementation.

**Comment by zedongh** [comment]
I do make a try (using pure rpc service api), seems that everything is fine. Cadence rpc core api does support activity and workflow be in different domains. 

Here are my analysis: 

Core api `RespondDecisionTaskCompleted` given `CompleteRequest` with `[]*Decision`

```go
RespondDecisionTaskCompleted(
    ctx context.Context,
    CompleteRequest *shared.RespondDecisionTaskCompletedRequest,
    opts ...yarpc.CallOption,
) (*shared.RespondDecisionTaskCompletedResponse, error)
```

`ScheduleActivityTaskDecisionAttributes` could specific `domain` field.

```go
type Decision struct {
	DecisionType                                             *DecisionType                                             `json:"decisionType,omitempty"`
	ScheduleActivityTaskDecisionAttributes                   *ScheduleActivityTaskDecisionAttributes                   
        ...
}

type ScheduleActivityTaskDecisionAttributes struct {
	ActivityId                    *string       `json:"activityId,omitempty"`
	ActivityType                  *ActivityType `json:"activityType,omitempty"`
	Domain                        *string       `json:"domain,omitempty"`
	TaskList                      *TaskList     `json:"taskList,omitempty"`
	Input                         []byte        `json:"input,omitempty"`
	...
}
```

**But cadence client doesn't support using `WithActivityOptions` given different domain (so always is nil value pass to server)**

Q: **Why I am trying to switch different domain in workflow?**
A: When user develop a new version worker wanna to deploy to production need do a small POC,  start workflow specify some activities of workflow could route to different domain (test/staging domain), without making a affect production workflow and worker.

**Reply by tfcace** [reply]
My preferred method is to put those Workflows and Activities on a separate task-list that's not part of any production worker.
That way, I can even test and debug it from my development machine.

**Reply by zedongh** [reply]
For some reasons, I use task-list to binding some external configuration (retrieve by SideEffect) . So in my scene switch between different domain much better.
